uppsrc/CtrlLib/Win32.cpp

```diff
@@ -206,14 +206,14 @@ bool FileSelNative::Execute(bool open, const char *dlgtitle) {
 				if(!IsNull(fn))
 				{
 					if(multi && fn.Find(' ') >= 0)
-						out << W32(String() << '\"' << fn << '\"');
+						out.Append( ToSystemCharsetW('\"' + fn + '\"'));
 					else
-						out << W32(fn);
-					out.Cat(0);
+						out.Append(ToSystemCharsetW(fn));
+					out.Add(0);
 				}
 			}
 		}
-		int l = min(out.GetLength(), bufsize - 1);
+		int l = min(out.GetCount()*2, bufsize - 1);
 		memcpy(buffer, out, l + 1);
```

Comment:
I feel like "l + 1" here might be wrong, in fact the whole handling of terminal double \0 does not feel right...

Reply:
Yes, you are absolutely right, thanks for your feelings :)
The "+ 1" is not needed, nor the "out.Add(0);" line.
I fixed in the branch.